packages/vgrammar-wordcloud/src/wordcloud.ts

```diff
@@ -169,13 +169,13 @@ const field = <T>(option: FieldOption | TagItemAttribute<T>) => {
   return (datum: any) => datum[(option as FieldOption).field] as T;
 };
 
+const sqrt = (x: number) => {
+  return x < 0 ? -Math.sqrt(-x) : Math.sqrt(x);
+};
+
 // 模拟sqrt scale
 const sqrtScale = (datum: any, domain: number[], range: number[]) => {
```

Comment:
maybe you can generate a function from `domain` and `range`, and there are two case need to be handle:
1. domain[0] === domain[1]
2. -domain[0] === domain[1] or domain[0] < 0

```
const generateSqrtScale = (domain: number[], range: number[]) => {
  if (domain[0] === domain[1]) {
        return (datum: number) => range[0];
  }
  if (domain[0] < 0) {
       return (datum: number) => ( ( sqrt(datum) / sqrt(domain[1] ) * (range[1] - range[0]) + range[0]);
  }
 
  return datum => ((sqrt(datum) - sqrt(domain[0])) / (sqrt(domain[1]) - sqrt(domain[0]))) * (range[1] - range[0]) + range[0];
   

}
```